src/java_bytecode/java_object_factory.h

```diff
@@ -16,6 +16,24 @@ exprt object_factory(
   const typet &type,
   code_blockt &init_code,
   bool allow_null,
-  symbol_tablet &symbol_table);
+  symbol_tablet &symbol_table,
+  size_t max_nondet_array_length,
+  const source_locationt &);
+
+void gen_nondet_init(
+  const exprt &expr,
+  code_blockt &init_code,
+  symbol_tablet &symbol_table,
+  const source_locationt &,
+  bool skip_classid = false,
```

Comment:
It's not good to use default values here. These are set by options, aren't they?